src/client/Microsoft.Identity.Client/ManagedIdentity/CredentialManagedIdentitySource.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    internal class CredentialManagedIdentitySource : AbstractManagedIdentity
```

Comment:
I'm debating about the name. If we think the CredentialManagedIdentity is going to apply to source eventually, then this is ok. But the more I think about it, the more I doubt it. So maybe ImdsV2 is better name?